tests/python/unittest/test_sparse_ndarray.py

```diff
@@ -918,18 +918,22 @@ def test_sparse_nd_check_format():
 
 @with_seed()
 def test_sparse_nd_norm():
-    def check_sparse_nd_norm(stype, shape, density):
+    def check_sparse_nd_norm(stype, shape, density, **kwargs):
         data, _ = rand_sparse_ndarray(shape, stype, density)
-        norm = data.norm()
-        expected_norm = np.linalg.norm(data.asnumpy())
-        assert_almost_equal(norm.asnumpy(), expected_norm)
+        norm = data.norm(**kwargs)
+        expected_norm = data.tostype('default').norm(**kwargs)
+        assert_almost_equal(norm.asnumpy(), expected_norm.asnumpy())
 
     shape = (5, 5)
     stypes = ['row_sparse', 'csr']
-    densities = [0, 0.5]
+    densities = [0, 0.5, 1]
     for stype in stypes:
         for density in densities:
-            check_sparse_nd_norm(stype, shape, density)
+           check_sparse_nd_norm(stype, shape, density, axis=None, keepdims=False, ord=2)
+
+    # test fallback
+    check_sparse_nd_norm(stype, shape, density, axis=0, keepdims=False, ord=2)
+    check_sparse_nd_norm(stype, shape, density, axis=None, keepdims=True, ord=2)
```

Comment:
i had this before but even for dense ord=2 is not implemented

Reply:
Typo: ord != 2

Reply:
okay. is there a plan to add backward pass test for l2 norm ?

Reply:
The sparse implementation doesn't support backward and it relies on the automatic fallback mechanism because no FInferStorage is not registered. For these kind of op we don't add extra unit tests. Automatic fallback is tested in  https://github.com/apache/incubator-mxnet/blob/master/tests/python/unittest/test_sparse_operator.py#L1619